### PR TITLE
nshlib: disable nsh error print if NSH_DISABLE_ERROR_PRINT enabled

### PR DESCRIPTION
**Pull request by anchao**
## Summary

nshlib: disable nsh error print if NSH_DISABLE_ERROR_PRINT enabled

```
   text    data     bss     dec     hex filename
 398953   27088    4128  430169   69059 nuttx    /* before */
 389241   27072    4128  420441   66a59 nuttx    /* after  */
  -9712     -16
```

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

sim/nsh

### CODE REVIEW

File: nshlib/nsh_console.c [hunk 1]
```diff
@@ -63,8 +63,10 @@ static int nsh_consoleioctl(FAR struct nsh_vtbl_s *vtbl,
                             int cmd, unsigned long arg);
 static int nsh_consoleoutput(FAR struct nsh_vtbl_s *vtbl,
                              FAR const char *fmt, ...) printf_like(2, 3);
+#ifndef CONFIG_DEFAULT_SMALL
 static int nsh_erroroutput(FAR struct nsh_vtbl_s *vtbl,
```
Comment (patacongo) [inline nshlib/nsh_console.c:67]: I would prefer that CONFIG_DEFAULT_SMALL not be used in this manner.  CONFIG_DEFAULT_SMALL is a property of the hardware (or should be), usually meaning is that the MCU has small memories.  Its function is to the choose a smaller default value for other settings and should not change for a given MCU.

My preference would be that you honor these semantics.  I suggest that add a new configuration like:

    config NSH_DISABLE_ERROR_PRINT
        bool "Disable NSH Error Printing"
       default y if DEFAULT_SMALL
       default n if !DEFAULT_SMALL

The option should be selected even if the MCU has CONFIG_DEFAULT_SMALL.

This comment applies to all of the size reduction changes you have made that uses CONFIG_DEFAULT_SMALL to directly control features.  It should not be used that way.

Comment (anchao) [reply nshlib/nsh_console.c:67]: Done, Thanks for review.

File: nshlib/Kconfig [hunk 2]
```diff
@@ -377,6 +377,11 @@ config NSH_DISABLE_HELP
 	bool "Disable help"
 	default DEFAULT_SMALL
 
+config NSH_DISABLE_ERROR_PRINT
+	bool "Disable NSH Error Printing"
+	default y if DEFAULT_SMALL
```
Comment (xiaoxiang781216) [inline nshlib/Kconfig:382]: default DEFAULT_SMALL

Comment (anchao) [reply nshlib/Kconfig]: Done

File: nshlib/nsh_console.c (hunk 1 above)
Comment (xiaoxiang781216) [reply nshlib/nsh_console.c:67]: @patacongo do you really mean CONFIG_SMALL_MEMORY not CONFIG_DEFAULT_SMALL?

File: nshlib/nsh_console.h [hunk 3]
```diff
@@ -58,6 +58,15 @@
 #  define nsh_output            vtbl->output
 #endif
 
+#ifdef CONFIG_NSH_DISABLE_ERROR_PRINT
+# undef nsh_error
```
Comment (xiaoxiang781216) [inline nshlib/nsh_console.h:62]: two space indent

File: nshlib/nsh_alias.c [hunk 4]
```diff
@@ -47,22 +47,14 @@
...
-
-/****************************************************************************
- * Private Data
- ****************************************************************************/
-
 /* Alias message format */
 
-static const char g_aliasfmt[]    = "alias %s='%s'\n";
-static const char g_savefailfmt[] = "alias %s='%s' failed\n";
+#define alias_format            "alias %s='%s'\n"
```
Comment (xiaoxiang781216) [inline nshlib/nsh_alias.c:52]: should we keep g_ prefix like g_dd?

Comment (pkarashchenko) [reply nshlib/nsh_alias.c:52]: Let's change to `atic const char *g_aliasfmt = "alias %s='%s'\n";` that will have basically the same effect with optimisation turned on

File: nshlib/nsh_console.h (hunk 3 above)
Comment (anchao) [reply nshlib/nsh_console.h]: Done

File: nshlib/nsh_alias.c (hunk 4 above)
Comment (anchao) [reply nshlib/nsh_alias.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

File: nshlib/nsh_alias.c (hunk 4 above)
Comment (pkarashchenko) [reply nshlib/nsh_alias.c]: So you still prefered to use define and not pointer to constant data? May I ask what is the difference in size/mem with define vs pointer to constant data?

Comment (xiaoxiang781216) [reply nshlib/nsh_alias.c]: The problem is that the compiler will complain static symbols isn't used.

**Pull request merged**